### PR TITLE
Clean network listener code

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 1]
```diff
@@ -118,16 +118,17 @@ class MainActivity : ThemedActivity() {
     }
 
     private fun observeNetworkStatus() {
+        var previousIsAvailable: Boolean? = null
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:121]: His trigger several times?

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:121]: Exactly, it triggers twice when the network is available.

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:121]: So this is not the place to manage it

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:121]: It is necessary to do it from the `livedata` itself, so it notifies only when it is necessary

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt]: At first, I thought the `LiveDataNetworkStatus` was bugged, but in fact it's not.
It broadcasts the global network status when any network's connectivity status changes.

So I just added a `distinctUntilChanged()` to only listen to the global network status changes.

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**